src/types/list.rs

```diff
@@ -132,6 +132,50 @@ impl PyList {
         }
     }
 
+    /// Deletes the `index`th element of self.
+    ///
+    /// This is equivalent to the Python statement `del self[i]`.
+    #[inline]
+    pub fn del_item(&self, index: usize) -> PyResult<()> {
+        unsafe {
+            err::error_on_minusone(
+                self.py(),
+                ffi::PySequence_DelItem(self.as_ptr(), get_ssize_index(index)),
+            )
+        }
+    }
+
+    /// Assigns the sequence `seq` to the slice of `self` from `low` to `high`.
+    ///
+    /// This is equivalent to the Python statement `self[low:high] = v`.
+    #[inline]
+    pub fn set_slice(&self, low: usize, high: usize, seq: &PyAny) -> PyResult<()> {
+        unsafe {
+            err::error_on_minusone(
+                self.py(),
+                ffi::PySequence_SetSlice(
```

Comment:
The CPython API provides `PyList_SetSlice` that may be optimized for list and may be worth using here.  `PyList_GetSlice` also exists but there is no  `PyList_DelSlice`.

Reply:
Good catch, from other methods I had assumed that all existing `PyList_` functions were already covered. I'll have to recheck!